fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJob.java

```diff
@@ -0,0 +1,409 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.load;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Range;
+import com.google.gson.annotations.SerializedName;
+import com.starrocks.analysis.InsertStmt;
+import com.starrocks.analysis.PartitionNames;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.OlapTable;
+import com.starrocks.catalog.Partition;
+import com.starrocks.catalog.PartitionInfo;
+import com.starrocks.catalog.PartitionKey;
+import com.starrocks.catalog.PartitionType;
+import com.starrocks.catalog.RangePartitionInfo;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.persist.AddPartitionsInfo;
+import com.starrocks.persist.InsertOverwriteStateChangeInfo;
+import com.starrocks.persist.PartitionPersistInfo;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.qe.QueryState;
+import com.starrocks.qe.StmtExecutor;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.sql.StatementPlanner;
+import com.starrocks.sql.plan.ExecPlan;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class InsertOverwriteJob {
```

Comment:
Can you split into two parts? A InsertOverwriteJobRunner and InsertOverwriteJobState.
The State will be persisted and the Runner would process a Job?

Reply:
Then you decouple the State and the Runner. When users create an Job, you create an State to represent it. and then when you are going to execute it, you will create an Runner.

Reply:
I split then original InsertOverwriteJob into two parts: InsertOverwriteJob and InsertOverwriteJobRunner. PTAL

Reply:
@imay PTAL